crates/driver/src/domain/competition/mod.rs

```diff
@@ -302,6 +337,70 @@ impl Competition {
         auction_id: Option<i64>,
         solution_id: u64,
         submission_deadline: u64,
+    ) -> Result<Settled, Error> {
+        let (response_tx, response_rx) = oneshot::channel();
+
+        let request = SettleRequest {
+            auction_id,
+            solution_id,
+            submission_deadline,
+            response_sender: response_tx,
+        };
+
+        self.settle_queue.try_send(request).map_err(|err| {
+            tracing::error!(?err, "Failed to enqueue /settle request");
+            Error::SubmissionError
+        })?;
+
+        response_rx.await.map_err(|err| {
+            tracing::error!(?err, "Failed to dequeue /settle response");
+            Error::SubmissionError
+        })?
+    }
+
+    async fn process_settle_requests(
+        self: Arc<Self>,
+        mut settle_rx: mpsc::Receiver<SettleRequest>,
+    ) {
+        while let Some(request) = settle_rx.recv().await {
+            let SettleRequest {
+                auction_id,
+                solution_id,
+                submission_deadline,
+                response_sender,
+            } = request;
+            if self.eth.current_block().borrow().number >= submission_deadline {
+                if let Err(err) = response_sender.send(Err(DeadlineExceeded.into())) {
+                    tracing::error!(
+                        ?err,
+                        "settle deadline exceeded. unable to return a response"
+                    );
+                }
+                continue;
+            }
+            let solver = self.solver.name().to_string();
+            let result = async {
+                observe::settling();
+                let result = self
+                    .process_settle_request(auction_id, solution_id, submission_deadline)
+                    .await;
+                observe::settled(self.solver.name(), &result);
+                result
+            }
+            .instrument(tracing::info_span!("/settle", solver, auction_id))
```

Comment:
Using `instrument()` is great. But ideally we would also instrument the warn log of the `DeadlineExceeded` case.
Seems like we could just throw all of the code inside this loop into a future and instrument that.